### PR TITLE
task/HW-62104 iOS Refactor LoginPresenter and LoginAuthenticationService

### PR DESCRIPTION
**Pull request by asenthil-altmtrk**

### CODE REVIEW

File: Tests/HyperwalletErrorTests.swift [hunk 1]
```diff
@@ -81,11 +81,11 @@ class HyperwalletErrorTests: XCTestCase {
     func testHyperwalletError_generalHttpError() {
         let hyperwalletError = HyperwalletError(message: "Please check your login credentials and try again",
                                                 code: "INCORRECT_LOGIN_CREDENTIALS")
-        let testErrorTypeHttp = HyperwalletErrorType.http(HyperwalletErrors(errorList: [hyperwalletError]), 401)
+        let testErrorTypeHttp = HyperwalletErrorType.http(HyperwalletErrors(errorList: [hyperwalletError]), 402)
```
Comment (jsaini-hw) [inline Tests/HyperwalletErrorTests.swift:84]: This should still be 401

File: Tests/HyperwalletErrorTests.swift [hunk 2]
```diff
@@ -81,11 +81,11 @@ class HyperwalletErrorTests: XCTestCase {
     func testHyperwalletError_generalHttpError() {
         let hyperwalletError = HyperwalletError(message: "Please check your login credentials and try again",
                                                 code: "INCORRECT_LOGIN_CREDENTIALS")
```
Comment (jsaini-hw) [inline Tests/HyperwalletErrorTests.swift:83]: This still qualifies for 401. Use 403 Use some other message like, "The caller does not have access to the requested resource" and code "FORBIDDEN"

Comment (asenthil-altmtrk) [reply Tests/HyperwalletErrorTests.swift]: Changed to FORBIDDEN

**Comment by jsaini-hw** [comment]
This is no longer required so closing it

**Pull request closed without merging**